src/vue/pages/IncomingWithdrawalRequests.vue

```diff
@@ -0,0 +1,368 @@
+        await Promise.all(
+          data
+            .map(request => new WithdrawalDetailsRequestRecord(request))
+            .map(async request => {
+              request.asset = await this.getAssetFromBalanceId(request)
+              return request
+            })
+        )
+      )
+    },
+
+    async getAssetFromBalanceId (request) {
+      const { _rawResponse: token } = await Api.api
+        .get(`balances/${request.balanceId}/asset`)
+      return token.data.code
+    },
+
+    showRequestDetails (index) {
+      this.selectedIndex = index
+      this.isDetailsDrawerShown = true
+    },
+
+    async approveRequest () {
+      this.isRequestRejected = true
+      try {
+        const action = base.xdr.ReviewRequestOpAction.approve().value
+        await this._reviewWithdraw({ action }, this.selectedRequest)
+        await this.rewriteWithdrawalRequest()
+
+        Bus.success('incoming-withdrawal-requests.request-approved-msg')
+      } catch (e) {
+        ErrorHandler.process(e)
+      }
+      this.isRequestRejected = false
+    },
+
+    async rejectRequest (rejectReason) {
+      this.isRequestRejected = true
+      try {
+        const action = base.xdr.ReviewRequestOpAction.permanentReject().value
+        /* eslint-disable-next-line max-len */
+        await this._reviewWithdraw({ action, reason: rejectReason }, this.selectedRequest)
+        /* eslint-enable-next-line max-len */
+        await this.rewriteWithdrawalRequest()
+
+        Bus.success('incoming-withdrawal-requests.request-rejected-msg')
+      } catch (e) {
+        ErrorHandler.process(e)
+      }
+      this.isRequestRejected = false
+    },
+
+    async _reviewWithdraw ({ action, reason = '' }, ...requests) {
+      const operations = requests.map(function (item) {
+        const opts = {
+          requestID: item.id,
+          requestHash: item.hash,
+          requestType: item.requestTypeI,
+          reviewDetails: {
+            tasksToAdd: 0,
+            tasksToRemove: item.pendingTasks,
+            externalDetails: '{}',
+          },
+          action,
+          reason,
+        }
+        return base.ReviewRequestBuilder.reviewWithdrawRequest({
+          ...opts,
+          requestDetails: JSON.stringify(opts),
+        })
+      })
+      await Api.api.postOperations(...operations)
+    },
+    async rewriteWithdrawalRequest () {
+      /* eslint-disable-next-line max-len */
+      const endpoint = `/${HORIZON_VERSION_PREFIX}/requests/${this.selectedRequest.id}`
+      /* eslint-enable-next-line max-len */
+      const { data } = await Api.api.getWithSignature(endpoint, {
+        filter: {
+          reviewer: this.accountId,
+        },
+        include: ['request_details'],
+      })
+      let withdraw = new WithdrawalDetailsRequestRecord(data)
+      withdraw.asset = await this.getAssetFromBalanceId(withdraw)
+
+      this.requestsHistory.splice(this.selectedIndex, 1, withdraw)
+    },
+  },
+}
+</script>
+
+<style lang="scss" scoped>
```

Comment:
We don't add `scoped` param for new components because of BEM

Reply:
resolved